Userland/Libraries/LibWeb/DOM/Document.cpp

```diff
@@ -4090,7 +4090,8 @@ void Document::ensure_animation_timer()
                 return;
             }
 
-            update_animations_and_send_events(window()->performance()->now());
+            if (window())
+                update_animations_and_send_events(window()->performance()->now());
```

Comment:
if document does not have window then probably it should not run animation driving timer. let's try to get to the root of the problem instead of covering it using null-check.

Reply:
Instead of doing this check, I've changed this to access `performance()` through the global object, which I believe should exist whether we have a window or not. I'm not 100% certain on this, but I think we still need to do this work, even if there is no window, due to animation events which are observable from Javascript.

Reply:
hmm, I would not expect animation events to be dispatched on documents without a window, considering these events are not dispatched even on elements with `display: none`. have you checked that?